spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/operations/SqsTemplate.java

```diff
@@ -408,10 +434,14 @@ private boolean isSkipAttribute(MessageSystemAttributeName name) {
 	}
 
 	private CompletableFuture<QueueAttributes> getQueueAttributes(String endpointName) {
-		return this.queueAttributesCache.computeIfAbsent(endpointName,
-				newName -> QueueAttributesResolver.builder().sqsAsyncClient(this.sqsAsyncClient).queueName(newName)
-						.queueNotFoundStrategy(this.queueNotFoundStrategy).queueAttributeNames(this.queueAttributeNames)
-						.build().resolveQueueAttributes());
+		return this.queueAttributesCache.computeIfAbsent(endpointName, newName -> {
+			// ensure we have the content based dedupe config to determine default fifo send headers
+			Set<QueueAttributeName> namesToRequest = new HashSet<>(queueAttributeNames);
+			namesToRequest.add(QueueAttributeName.CONTENT_BASED_DEDUPLICATION);
+			return QueueAttributesResolver.builder().sqsAsyncClient(this.sqsAsyncClient).queueName(newName)
```

Comment:
I think we should do this only for Fifo queues.

In order to keep responsibilities segregated, I'm thinking we can create an overload for `getQueueAttributes(String endpointName, Collection<String> additionalAttributes) ` and only call it from the Fifo methods.

How does this sound?